Providers/Profile.swift

```diff
@@ -949,8 +949,16 @@ open class BrowserProfile: Profile {
 
         fileprivate func syncClientsWithDelegate(_ delegate: SyncDelegate, prefs: Prefs, ready: Ready, why: SyncReason) -> SyncResult {
             log.debug("Syncing clients to storage.")
+
+            func notifyAll() {
+                guard let account = self.profile.getAccount() else {
+                    return
+                }
+                account.notifyAll(collectionsChanged: ["clients"])
+            }
+
             let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs, why: why)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info) >>== { result in
+            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info, onFirstSync: notifyAll) >>== { result in
```

Comment:
I really don't like the `onFirstSync` idea. It's an odd bidirectional leak: you're anticipating some internal knowledge of the synchronizer (timestamp is zero implies uploading our record), and you're leaking a side-effect into the synchronizer.

I suggest you think about what you actually want to do here. Do you want to notify other clients if the `clients` collection changed? After all, that's what we want to do for sending a command and uploading our record, right?

If so, look at the stats the method returns, and do the work here!